src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -337,40 +336,35 @@ bool primitive_inst::update_impl() {
         }
         if (!has_cached_impl) {
             if (_dynamic_impl) {
-                auto& compilation_context = get_network().get_compilation_context();
-                compilation_context.push_task(impl_key, [this, updated_params, impl_key](kernels_cache& kc) {
-                    auto& cache = get_network().get_implementations_cache();
-                    {
-                        std::lock_guard<std::mutex> lock(get_network().get_impl_cache_mutex());
-                        // Check existense in the cache one more time as several iterations of model execution could happens and multiple compilation
-                        // tasks created for same shapes
-                        if (cache.has(impl_key))
-                            return;
+                auto& compilation_context = get_network().get_program()->get_compilation_context();
+                compilation_context.push_task(impl_key, [this, &compilation_context, updated_params, impl_key]() {
+                    if (compilation_context.is_stopped())
+                        return;
+                    try {
+                        auto _program = get_network().get_program();
+                        auto& cache = _program->get_implementations_cache();
+                        {
+                            // Check existense in the cache one more time as several iterations of model execution could happens and multiple compilation
+                            // tasks created for same shapes
+                            if (cache.has(impl_key))
+                                return;
+                        }
+
+                        auto impl = _node->type()->choose_impl(*_node, updated_params);
+                        impl->build_kernels(_program->get_kernels_cache());
+                        cache.add(impl_key, impl->clone());
+                    } catch (std::exception& ex) {
+                        std::cout << " Exception " << ex.what() << std::endl;
```

Comment:
Please use OPENVINO_ASSERT instead of cout.

Reply:
Removed it. It is redundant code only for debug issue. I forget to remove codes. thanks.